File: DemoServer/AdditionalLanguages/go/Basics/editDocument/EditDocument.go
package main

//region Usings
import "github.com/ravendb/ravendb-go-client"
//endregion

var globalDocumentStore *ravendb.DocumentStore

func main() {
    createDocumentStore()
    editDocument("newCompanyName")
    globalDocumentStore.Close()
}

func createDocumentStore() (*ravendb.DocumentStore, error) {
    if globalDocumentStore != nil {
        return globalDocumentStore, nil
    }
    urls := []string{"http://localhost:8080"}
    store := ravendb.NewDocumentStore(urls, "testGO")
    err := store.Initialize()
    if err != nil {
        return nil, err
    }
    globalDocumentStore = store
    return globalDocumentStore, nil
}

type Company struct {
    ID      string
    Name    string
    Phone   string
    Contact *Contact
}

type Contact struct {
    Name  string
    Title string
}

//region Demo
func editDocument(companyName string) error {

    session, err := globalDocumentStore.OpenSession("")
    if err != nil {
        return err
    }
    defer session.Close()

    //region Step_1
    var company *Company  
    err = session.Load(&company, "companies/5-A")
    if err != nil {
        return err
    }   
    
    if company == nil {
        return nil
    }
    //endregion
    
    //region Step_2
    company.Name = companyName
    //endregion    

    //region Step_3
    err = session.SaveChanges()
    if err != nil {
        return err
    }
    //endregion

    return nil
}
//endregion

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/basics/editDocument/EditDocument.java
package net.ravendb.demo.basics.editDocument;

//region Usings
import net.ravendb.client.documents.session.IDocumentSession;
//endregion
import net.ravendb.demo.common.DocumentStoreHolder;
import net.ravendb.demo.common.models.Company;

public class EditDocument {

    public void run(RunParams runParams) {
        String companyName = runParams.getCompanyName();

        //region Demo
        try (IDocumentSession session = DocumentStoreHolder.store.openSession()) {
            //region Step_1
            Company company = session.load(Company.class, "companies/5-A");
            //endregion

            //region Step_2
            company.setName(companyName);
            //endregion

            //region Step_3
            session.saveChanges();
            //endregion
        }
        //endregion
    }

    public static class RunParams {
        private String companyName;

        public String getCompanyName() {
            return companyName;
        }

        public void setCompanyName(String companyName) {
            this.companyName = companyName;
        }
    }
}
